scripts/apishow.py

```diff
@@ -1,55 +1,235 @@
+    except IndexError:
+        subcollections = {}
+
+    def _process_collection(collection, store):
+        _init_store(collection.name, store)
+
+        try:
+            actions = collection.action.all
+        except KeyError:
+            actions = []
+
+        if 'actions_avail' not in store[collection.name]:
+            store[collection.name]['actions_avail'] = set(actions)
+
+        try:
+            collection_len = len(collection)
+        except AttributeError:
+            collection_len = 0
+
+        if collection_len > 0:
+            _init_store('entity', store[collection.name])
+            entity = random.choice(collection)
+
+            try:
+                actions = entity.action.all
+            except KeyError:
+                actions = []
+            if 'actions_avail' not in store[collection.name]['entity']:
+                store[collection.name]['entity']['actions_avail'] = set(actions)
+
+            if collection.name in subcollections:
+                for subcol_name in subcollections[collection.name]:
+                    try:
+                        subcol = getattr(entity, subcol_name)
+                    except AttributeError:
+                        continue
+                    _process_collection(subcol, store[collection.name])
+
+    for collection in api.collections.all:
+        _process_collection(collection, store)
+
+
+def print_info(store):
+    """Print info about collections together with coverage info when available."""
+
+    for name, collection in sorted(store.iteritems()):
+        print('=' * (2 + len(name)))
+        print('* {}'.format(name))
+
+        def _print_resource(res_title, res_dict):
+            if 'actions_avail' in res_dict and res_dict['actions_avail']:
+                print('  {} actions:'.format(res_title))
+                covered = True if 'actions' in res_dict else False
+                for action in res_dict['actions_avail']:
+                    print('    * {}{}'.format(
+                        action,
+                        ' OK' if covered and action in res_dict['actions'] else ''))
+            if 'entity' in res_dict:
+                _print_resource('{} entity'.format(res_title), res_dict['entity'])
+
+            for key, subcollection in sorted(res_dict.iteritems()):
+                if key in ('actions', 'actions_avail', 'entity', 'methods'):
+                    continue
+                _print_resource('Subcollection "{}"'.format(key), subcollection)
+
+        _print_resource('Collection', collection)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        description=__doc__, formatter_class=argparse.RawDescriptionHelpFormatter)
+    parser.add_argument(
+        '--address',
+        default=conf.env.get('base_url', None),
+        help="hostname or ip address of target appliance")
+    parser.add_argument(
+        '--logfile',
+        metavar='FILE',
+        default=None,
```

Comment:
@mshriver good idea, I originally wanted the coverage to be optional, but since parsing the logfile has negligible overhead compared to getting all the collections/subcollections/entities info out of the REST API client, it can be on by default.